ReactAndroid/src/main/java/com/facebook/react/bridge/UIManagerListener.java

```diff
@@ -12,10 +12,33 @@ public interface UIManagerListener {
   /**
    * Called right before view updates are dispatched at the end of a batch. This is useful if a
    * module needs to add UIBlocks to the queue before it is flushed.
+   *
+   * This is called by Paper only.
```

Comment:
NAB for me, but I wonder if when you go to upstream they'll want you to create separate subclasses of UIManagerListener for Paper v.s. Fabric

Reply:
Or rather sub-interface ... I'm out of practice with Java but I'm pretty sure interfaces can `extend` other interfaces? So you'd end up having `BaseUIManagerListener` and `PaperUIManagerListener` and `FabricUIManagerListener`

Reply:
I think it is the same interface because this is used in an interface that abstracts UIManager and FabricUIManager. See https://github.com/facebook/react-native/blob/main/ReactAndroid/src/main/java/com/facebook/react/bridge/UIManager.java#L110. I also agree that it is probably better as separate interfaces, but might be better to leave as is for now.